nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/JoltTransformJSON.java

```diff
@@ -147,10 +174,39 @@
                         .explanation(message)
                         .build());
             }
+
         } else {
+
+            ClassLoader customClassLoader = null;
             try {
+
+                if(modulePath != null) {
+                    customClassLoader = ClassLoaderUtils.getCustomClassLoader(modulePath, this.getClass().getClassLoader(), getJarFilenameFilter());
+                }
+
                 Object specJson = SORTR.getValue().equals(transform) ? null : JsonUtils.jsonToObject(specValue, DEFAULT_CHARSET);
-                TransformFactory.getTransform(transform, specJson);
+
+                if(CUSTOMR.getValue().equals(transform)){
+
+                    if(!StringUtils.isEmpty(customTransform) && customClassLoader == null){
+                        final String customMessage = "One or more module directories containing the provided custom transformation must be specified.";
```

Comment:
See above about the message for whether a module directory MUST be provided or not